### DISCUSSION TITLE
Build Errors with WatchOS Target

### DISCUSSION THREAD

**Discussion by cengelbart39** [skiptools/skip#192] [Q&A] [answered]
I am adding an Apple Watch target to my project and intend to use the same Skip-supported packages between the main and Watch targets.

When I add my package under "Frameworks, Libraries, and Embedded Content", I get a bunch of build errors like:
`Multiple commands produce '/Users/christopher/Library/Developer/Xcode/DerivedData/MyApp-brdkhrkgdmszoccikqktuftwxucu/SourcePackages/plugins/skip-foundation.output/SkipFoundation/skipstone/.SkipFoundation.sourcehash'` 

I get one per package that uses the Skip Transpiler, including my packages. I would imagine both the iOS and watchOS targets are trying to create the sourcehash, hence the error message.

Any help would be appreciated.

`skip checkup` output if relevant:
```
[✓] Skip version 1.0.4 (= 1.0.4)
[✓] macOS version 14.6.1 (> 13.5.0)
[✓] macOS architecture: ARM
[✓] Swift version 5.10 (> 5.9.0)
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 5
[✓] Homebrew version 4.3.15 (> 4.1.0)
[✓] Gradle version 8.9 (> 8.6.0)
[✓] Java version 22.0.2 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.1
[✓] Android tools SDKs: 4
[✓] Resolve dependencies (9.74s)
[✓] Build hello-skip (10.02s)
[✓] Test Swift (5.71s)
[✓] Test Kotlin (21.29s)
[✓] Archive iOS ipa (14.15s)
[✓] Assemble HelloSkip-release.ipa 37 KB
[✓] Verify HelloSkip-release.ipa 37 KB
[✓] Assembling Android apk (76.58s)
[✓] Verify HelloSkip-release.apk 13 MB
[✓] Check Swift Package (0.19s)
[✓] Check Skip Updates: 1.0.4
[✓] Skip 1.0.4 checkup succeeded in 138.5s
```

**Comment by marcprux** [comment]
Is this a new stand-alone WatchOS app, or is it a companion app for an existing iOS App? Should you show us a screenshot of the Xcode target configuration of the app? It sounds like it might be including Skip's top-level target twice somehow, which would explain that error.

**Reply by cengelbart39** [reply]
It's a companion app for an existing iOS app, although it can run independently of the iOS app.

To get the Watch target information to populate correctly, I only use the xcconfig on the main app target, otherwise, the Watch target would be locked to iOS and macOS.
<img width="559" alt="Screenshot 2024-08-15 at 7 55 49 PM" src="https://github.com/user-attachments/assets/1dc6c09a-3e80-4ad8-ace6-43325e8a0fe0">

This is the general tab for the target in Xcode. `RainbowUI` is one of the modules created for the project, depending on two more custom modules.
<img width="674" alt="Screenshot 2024-08-15 at 8 06 27 PM" src="https://github.com/user-attachments/assets/dd24fe6a-a4d9-4297-8991-a761e231f3bd">

If I remove `RainbowUI` from the libraries list, it gets rid of those errors, although then it can't find `RainbowUI` or its dependencies. Nor is this a problem in the Mac build either.

**Comment by marcprux** [comment, marked as answer]
I'm seeing this error too when I add a watchOS target to the default HelloSkip app. Unfortunately, it seems to be a known issue with plugins and watchOS targets: https://github.com/mac-cain13/R.swift/issues/815 (found via https://stackoverflow.com/a/76030048/10253094). 

Unfortunately, I can't think of any good workaround that doesn't involve duplicating the target with one that shares the code but doesn't invoke the transpiler. You might be able to get creative with symbolic links between source files in different targets, but I'll see if I can think of a better solution. Note that this *might* be fixed by Xcode 16 – since I know they refactored how plugin targets are structured somewhat – but we are currently having other (unrelated) issues running with the latest beta, so I'm not sure I would advise investigating that yet.

**Reply by cengelbart39** [reply]
I came across that as well, but figured I'd try here anyway.

Thanks for looking into it at least.
